### DISCUSSION TITLE
Unable to locate 'Avalonia.Platform.IPlatformRenderInterface'.

### DISCUSSION THREAD

**Discussion by CzBuCHi** [AvaloniaUI/Avalonia#12805] [Q&A] [answered]
**Describe the bug**
Cannot create `Avalonia.Media.Imaging.Bitmap` from stream.

**To Reproduce**

Code:
```
var stream = File.OpenRead("path_to_image.png");
new Avalonia.Media.Imaging.Bitmap(stream);
```

Exception: `Unable to locate 'Avalonia.Platform.IPlatformRenderInterface'.` stack trace:
```
at Avalonia.LocatorExtensions.GetRequiredService[T](IAvaloniaDependencyResolver resolver)
at Avalonia.Media.Imaging.Bitmap.GetFactory()
at Avalonia.Media.Imaging.Bitmap..ctor(Stream stream)
at ProjectsManager.Services.BrowserService.GetIconFromPathAsBitmap(String filePath) in C:\projects\ProjectsManager\ProjectsManager\Services\BrowserService.cs:line 78

```
**Desktop**

- OS: Windows 11 22H2 Build 22621.2134
- Version 11.0.4

**Comment by Gillibald** [comment]
You are most likely trying to create an Avalonia Bitmap without initializing an Avalonia Application

**Comment by CzBuCHi** [comment, marked as answer]
i tryied to create bitmap in service ctor that is registered in bootstraper:
```
 public static AppBuilder BuildAvaloniaApp() {
        Bootstrapper.RegisterAsync(Locator.CurrentMutable, Locator.Current).Wait();
        return AppBuilder.Configure<App>()
                         .UsePlatformDetect()
                         .WithInterFont()!
                         .LogToTrace()
                         .UseReactiveUI();
    }
```

just tryied to create bitmap in binding valueconverter and it worked ...

reason for this report was that there is absolutely no info about `IPlatformRenderInterface` and i didnt find any class that implements it ...

**Comment by timunie** [comment]
@CzBuCHi can we convert this issue into a Q&A?

**Comment by CzBuCHi** [comment]
> can we convert this issue into a Q&A?

sure ... but i think better would be to add paragraph in documentation about `Avalonia.Media.Imaging.Bitmap` and similar that have those invisible dependencies

**Comment by timunie** [comment]
> but i think better would be to add paragraph in documentation about Avalonia.Media.Imaging.Bitmap and similar that have those invisible dependencies

can be done in addition for sure. But this issue (or even better this PR) should go here: https://github.com/AvaloniaUI/avalonia-docs
